install/playbooks/roles/borg-backup/files/backup.py

```diff
@@ -151,53 +151,29 @@ def loadKey(self, path):
+        # When using SSH, we do not need to mount remote SSH location.
+        # for this scheme, it is assumed that the remote server has borg
+        # installed otherwise, use sshfs://
         if self.location.scheme == 'ssh':
-            self.repositoryPath = self.url[6:]
-            self.repositoryMounted = True
+            self.repositoryPath = self.url[6:] + '/@server'
+            self.repositoryMounted = False
             return True
 
-        # Make sure the directory to mount the backup exists
-        self.mountPath = '/mnt/backup/' + self.configName
-        os.makedirs(self.mountPath, exist_ok=True)
-
-        # The backup location can be a remote directory mounted locally
-        # or even a local partition. Removing dir://
+        # The dir:// backup location can be a remote directory mounted locally
+        # or even a local partition.
         if self.location.scheme == 'dir':
-            self.repositoryPath = self.location.path
-            self.repositoryMounted = True
+            self.repositoryPath = self.location.path + '/@server'
+            self.repositoryMounted = False
+            os.makedirs(self.location.path, exist_ok=True)
             return True
 
-        # The location is a USB device mounted automatically using systemd
-        if self.location.scheme == 'usb':
-            self.repositoryPath = self.location.path
-            self.mountPath = '/mnt/backup/' + self.configName
-            self.repositoryMounted = os.path.ismount(self.mountPath)
-            return self.repositoryMounted
-
-        # The location is an S3 bucket mounted automatically using systemd
-        if self.location.scheme == 's3fs':
-            self.repositoryPath = self.location.path
+        # These locations are mounted automatically using systemd
+        if self.location.scheme in {'usb', 's3fs', 'sshfs'}:
+            # Make sure the directory to mount the backup exists
             self.mountPath = '/mnt/backup/' + self.configName
-            self.repositoryMounted = os.path.ismount(self.mountPath)
-            return self.repositoryMounted
-
-        ## When using SSHFS, the remote location is mounted using fuse
-        # In this case, the mount path and the repository path are the same.
-        if self.location.scheme == 'sshfs':
-            self.repositoryPath = '/mnt/backup/' + self.configName
-            self.mountPath = '/mnt/backup/' + self.configName
-            self.repositoryMounted = True
-            return True
-
-        # Check if the directory is already mounted
-        # In this case, we return true, and we will expect the
-        # next functions to check if this is a valid repository
-        if os.path.ismount(self.mountPath):
-            self.repositoryPath = self.mountPath
+            self.repositoryPath = self.mountPath + '/@server'
             self.repositoryMounted = True
+            os.makedirs(self.mountPath, exist_ok=True)
             return True
```

Comment:
This might be a mistake.
This was the behavior for remote locations:
```
os.makedirs(self.mountPath, exist_ok=True)
self.repositoryMounted = os.path.ismount(self.mountPath)
return self.repositoryMounted
```
and it might be more appropriate.

Reply:
I think I was returning true in case of local backups, if this is a directory, considering it as always "mounted". I am using it so far without issue, but _maybe_ it could be cleaner to use the mounted option only for "mountable" backup locations.

Reply:
Yes, the `repositoryMounted` flag is only used to unmount when needed. So for local backups, false is ok and the unmount is not called. There was a check in the unmount function for local backups that is removed also.